teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt

```diff
@@ -3,27 +3,18 @@ package org.jetbrains.teamcity.rest
 import java.io.File
 import java.util.*
 
-abstract class TeamCityInstance {
-    abstract fun withLogResponses() : TeamCityInstance
-
-    abstract fun builds(): BuildLocator
-
-    abstract fun build(id: BuildId): Build
-    abstract fun build(buildType: BuildConfigurationId, number: String): Build
-    abstract fun buildConfiguration(id: BuildConfigurationId): BuildConfiguration
-    abstract fun vcsRoots(): VcsRootLocator
-    abstract fun vcsRoot(id: VcsRootId): VcsRoot
-    abstract fun project(id: ProjectId): Project
-    abstract fun rootProject(): Project
-
-    companion object {
-        @JvmStatic
-        fun guestAuth(serverUrl: String): TeamCityInstance = createGuestAuthInstance(serverUrl)
-
-        @JvmStatic
-        fun httpAuth(serverUrl: String, username: String, password: String): TeamCityInstance
-                = createHttpAuthInstance(serverUrl, username, password)
-    }
+interface TeamCityApi {
```

Comment:
Yeah, we can leave this in the API module. But I'd still like to have interface in API (not an abstract class) with deprecated companion object